gson/src/main/resources/META-INF/proguard/gson.pro

```diff
@@ -15,25 +15,20 @@
 # Note: Cannot perform finer selection here to only cover Gson annotations, see also https://stackoverflow.com/q/47515093
 -keepattributes RuntimeVisibleAnnotations,AnnotationDefault
 
-
 ### The following rules are needed for R8 in "full mode" which only adheres to `-keepattribtues` if
 ### the corresponding class or field is matches by a `-keep` rule as well, see
 ### https://r8.googlesource.com/r8/+/refs/heads/main/compatibility-faq.md#r8-full-mode
 
-# Keep class TypeToken (respectively its generic signature)
--keep class com.google.gson.reflect.TypeToken { *; }
+# Keep class TypeToken (respectively its generic signature) if present
+-if class com.google.gson.reflect.TypeToken
+-keep,allowobfuscation class com.google.gson.reflect.TypeToken
 
 # Keep any (anonymous) classes extending TypeToken
 -keep,allowobfuscation class * extends com.google.gson.reflect.TypeToken
 
 # Keep classes with @JsonAdapter annotation
 -keep,allowobfuscation,allowoptimization @com.google.gson.annotations.JsonAdapter class *
 
-# Keep fields with @SerializedName annotation, but allow obfuscation of their names
--keepclassmembers,allowobfuscation class * {
-  @com.google.gson.annotations.SerializedName <fields>;
-}
```

Comment:
Out of curiosity, why did you remove this here and add it below guarded by a `-if class *`? Does that make a difference for the end result?

Reply:
Yes, in R8 full mode we have a distinction between referenced and instantiated classes. For classes which are not seen by R8 as instantiated (e.g. if only referenced statically using `.class` as one will do in `fromJson` then `-keepclassmembers` will only keep static members and not instance members. For ProGuard and R8 in compatability mode the two will do the same.

Reply:
Hmm thanks, ok that sounds like the previous rules would not have worked properly then, though I think the shrinker-test showed that it did work as expected?
I am bit confused, or am I misunderstanding your comment or are the shrinker-tests flawed and not close enough to real code?

Reply:
One way to see the difference is to have an instance field annotated with `@SerializedName` which is not used in the code at all, then the field will be removed with the previous rule in R8 full mode (which is default from AGP 8.0.0). Then in a pass through from `fromJson` to `toJson` the field value will be lost. Of course it is a matter of taste if unused fields should be kept this way.